source/MaterialXGenShader/ShaderGraph.cpp

```diff
@@ -519,18 +533,39 @@ ShaderNode* ShaderGraph::addNode(const Node& node, ShaderGenerator& shadergen)
         }
     }
 
+    const string& targetColorSpace = _document ? _document->getAttribute(Element::COLOR_SPACE_ATTRIBUTE) : EMPTY_STRING;
+    for (InputPtr input : node.getInputs())
+    {
+        populateInputColorTransformMap(node, newNode, input, targetColorSpace);
+    }
+    for (ParameterPtr parameter : node.getParameters())
+    {
+        populateInputColorTransformMap(node, newNode, parameter, targetColorSpace);
+    }
+
     // Check if this is a file texture node that requires color transformation.
     if (newNode->hasClassification(ShaderNode::Classification::FILETEXTURE))
     {
         ParameterPtr file = node.getParameter("file");
```

Comment:
Maybe do nothing here if file is null, and then you don't have to check file multiple times below.